extensions/undertow/runtime/src/main/java/io/quarkus/undertow/runtime/UndertowHandlersConfServletExtension.java

```diff
@@ -0,0 +1,51 @@
+package io.quarkus.undertow.runtime;
+
+import java.io.InputStream;
+import java.util.List;
+
+import javax.servlet.ServletContext;
+
+import io.undertow.Handlers;
+import io.undertow.server.HandlerWrapper;
+import io.undertow.server.HttpHandler;
+import io.undertow.server.handlers.builder.PredicatedHandler;
+import io.undertow.server.handlers.builder.PredicatedHandlersParser;
+import io.undertow.servlet.ServletExtension;
+import io.undertow.servlet.api.DeploymentInfo;
+
+/**
+ * Registers the Undertow handlers configured in the app's META-INF/undertow-handlers.conf
+ */
+public class UndertowHandlersConfServletExtension implements ServletExtension {
+
+    private static final String META_INF_UNDERTOW_HANDLERS_CONF = "META-INF/undertow-handlers.conf";
+
+    public static boolean existsConfFile() {
+        ClassLoader classLoader = Thread.currentThread().getContextClassLoader();
```

Comment:
I am not sure about this, this can read it from any dependency. As we are mostly just interested in one actually present in the app I think you should be looking at ArchiveRootBuildItem to just check in the main application. 

This should also be registered under a HotDeploymentWatchedFileBuildItem so that changes to the file will cause an application restart.

Reply:
I have no idea of what you're talking about but that seems to make complete sense :smile: 
Can you point me to some code that uses this feature? 
Also I promise I won't be offended if you reject this PR and file another (decent) one 😁